data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/parser/LogstashConfigConverter.java

```diff
@@ -0,0 +1,56 @@
+package org.opensearch.dataprepper.logstash.parser;
+
+import com.amazon.dataprepper.model.configuration.PipelineModel;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.opensearch.dataprepper.logstash.LogstashLexer;
+import org.opensearch.dataprepper.logstash.LogstashParser;
+import org.opensearch.dataprepper.logstash.mapping.LogstashMapper;
+import org.opensearch.dataprepper.logstash.model.LogstashConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Converts Logstash configuration file and returns YAML file location
+ *
+ * @since 1.2
+ */
+public class LogstashConfigConverter {
+    public String convertLogstashConfigurationToPipeline(String logstashConfigurationPath, String outputDirectory) throws IOException {
+        final String logstashConfigAsString = new String(Files.readAllBytes(Paths.get(logstashConfigurationPath)));
+
+        LogstashLexer lexer = new LogstashLexer(CharStreams.fromString(logstashConfigAsString));
+        CommonTokenStream tokens = new CommonTokenStream(lexer);
+        LogstashParser parser = new LogstashParser(tokens);
+        ParseTree tree = parser.config();
+
+        LogstashVisitor visitor = new LogstashVisitor();
+        LogstashConfiguration logstashConfiguration = (LogstashConfiguration) visitor.visit(tree);
+
+        LogstashMapper logstashMapper = new LogstashMapper();
+        PipelineModel pipelineModel = logstashMapper.mapPipeline(logstashConfiguration);
+        Map<String, Object> pipeline = Collections.singletonMap("log-pipeline", pipelineModel);
```

Comment:
We don't know if the pipeline is a `log-pipeline` or not. I'd be interested in using the filename as the pipeline name. For example a file named `local-log-ingest.yaml` might be given the name `local-log-ingest`.

If we don't take that approach, maybe use a constant value like `logstash-converted-pipeline`.